ecosystem/indexer-grpc/indexer-grpc-fullnode-table-info/src/convert.rs

```diff
@@ -0,0 +1,837 @@
+                mutable: *mutable,
+                to: Some(Box::new(convert_move_type(to))),
+            }),
+        )),
+        MoveType::Unparsable(string) => {
+            Some(transaction::move_type::Content::Unparsable(string.clone()))
+        },
+    };
+    transaction::MoveType {
+        r#type: r#type as i32,
+        content,
+    }
+}
+
+#[inline]
+pub fn convert_write_set_changes(changes: &[WriteSetChange]) -> Vec<transaction::WriteSetChange> {
+    changes.iter().map(convert_write_set_change).collect()
+}
+
+#[inline]
+pub fn convert_hex_string_to_bytes(hex_string: &str) -> Vec<u8> {
+    hex::decode(hex_string.strip_prefix("0x").unwrap_or(hex_string))
+        .unwrap_or_else(|_| panic!("Could not convert '{}' to bytes", hex_string))
+}
+
+pub fn convert_move_struct_tag(struct_tag: &MoveStructTag) -> transaction::MoveStructTag {
+    transaction::MoveStructTag {
+        address: struct_tag.address.to_string(),
+        module: struct_tag.module.to_string(),
+        name: struct_tag.name.to_string(),
+        generic_type_params: struct_tag
+            .generic_type_params
+            .iter()
+            .map(convert_move_type)
+            .collect(),
+    }
+}
+
+pub fn convert_delete_module(delete_module: &DeleteModule) -> transaction::DeleteModule {
+    transaction::DeleteModule {
+        address: delete_module.address.to_string(),
+        state_key_hash: convert_hex_string_to_bytes(&delete_module.state_key_hash),
+        module: Some(transaction::MoveModuleId {
+            address: delete_module.module.address.to_string(),
+            name: delete_module.module.name.to_string(),
+        }),
+    }
+}
+
+pub fn convert_delete_resource(delete_resource: &DeleteResource) -> transaction::DeleteResource {
+    transaction::DeleteResource {
+        address: delete_resource.address.to_string(),
+        state_key_hash: convert_hex_string_to_bytes(&delete_resource.state_key_hash),
+        r#type: Some(convert_move_struct_tag(&delete_resource.resource)),
+        type_str: delete_resource.resource.to_string(),
+    }
+}
+
+pub fn convert_write_set_change(change: &WriteSetChange) -> transaction::WriteSetChange {
+    fn retry_get_data_from_table_item(
+        write_table_item: &WriteTableItem,
+    ) -> Option<DecodedTableData> {
+        let mut retried = 0;
+        while retried < QUERY_RETRIES {
+            retried += 1;
+            if let Some(data) = write_table_item.data.as_ref() {
+                return Some(data.clone());
+            }
+            std::thread::sleep(std::time::Duration::from_millis(QUERY_RETRY_DELAY_MS));
+        }
+        None
+    }
+    fn retry_get_data_from_delete_table_item(
+        delete_table_item: &DeleteTableItem,
+    ) -> Option<DeletedTableData> {
+        let mut retried = 0;
+        while retried < QUERY_RETRIES {
+            retried += 1;
+            if let Some(data) = delete_table_item.data.as_ref() {
+                return Some(data.clone());
+            }
```

Comment:
do we need to retry when parsing from api txn to pb? will api txn get overridden?